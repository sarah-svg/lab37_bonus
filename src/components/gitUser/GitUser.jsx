import React from 'react';
import PropTypes from 'prop-types';

function GitUser({  username, followers, following, url }) {
  return (
    <div>
      <p> {username}</p>
      <p>{followers}</p>
      <p>{following}</p>
      <p>{url}</p>
    </div>
  );
}

GitUser.propTypes = {
  username: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default GitUser;

