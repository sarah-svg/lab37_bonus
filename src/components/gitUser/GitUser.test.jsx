import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import GitUser from './GitUser';


describe('git user component', () => {
  it('renders to git user page', () => {
    afterEach(() => cleanup());
    const { asFragment } = render(
      <MemoryRouter> <GitUser username={'sarah-svg'} /></MemoryRouter> 
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
