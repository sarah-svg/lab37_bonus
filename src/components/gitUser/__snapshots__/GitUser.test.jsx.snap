// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git user component renders to git user page 1`] = `
<DocumentFragment>
   
  <div>
    <p>
       sarah-svg
    </p>
    <p />
    <p />
    <p />
  </div>
</DocumentFragment>
`;
